samples/apply-blend-renderer-to-hillshade/src/main/java/com/esri/arcgismaps/sample/applyblendrenderertohillshade/components/ApplyBlendRendererToHillshadeViewModel.kt

```diff
@@ -0,0 +1,231 @@
+    val colorRampPresets = listOf("None", "DEM Light", "Screen Display", "Elevation")
+
+    // Index of the selected color ramp preset; 0 means "None" (no color ramp).
+    var selectedColorRampPresetIndex by mutableStateOf(0)
+        private set
+
+    init {
+        // Load the raster layers and load and set up the map. Perform long-running loads in a coroutine
+        // and surface any errors through the MessageDialogViewModel.
+        viewModelScope.launch {
+            try {
+                // Load both raster layers. If any fails, report the error.
+                elevationBasemap.load().onFailure { throw it }
+                imageryBasemap.load().onFailure { throw it }
+                updateRenderer()
+                arcGISMap.load().onFailure { throw it }
+            } catch (ex: Throwable) {
+               messageDialogVM.showMessageDialog(ex)
+            }
+        }
+    }
+
+    /**
+     * Constructs and applies a BlendRenderer
+     * using the current UI parameters. Also switches the map basemap to the elevation
+     * basemap when a color ramp is active so the color ramp is visible.
+     */
+    private fun updateRenderer() {
+        try {
+            val colorRamp = createColorRampFromPresetIndex(selectedColorRampPresetIndex)
+
+            if (colorRamp != null) {
+                // When a color ramp is applied, set the basemap to the elevation raster
+                val renderer = BlendRenderer(
+                    elevationRaster = elevationRaster,
+                    outputMinValues = listOf(9.0),
+                    outputMaxValues = listOf(255.0),
+                    sourceMinValues = emptyList(),
+                    sourceMaxValues = emptyList(),
+                    noDataValues = emptyList(),
+                    gammas = emptyList(),
+                    colorRamp = colorRamp,
+                    altitude = altitude,
+                    azimuth = azimuth,
+                    slopeType = slopeType ?: SlopeType.None
+                )
+                elevationRasterLayer.renderer = renderer
+                arcGISMap.setBasemap(elevationBasemap)
+            } else {
+                // Since no ColorRamp is selected, apply BlendRenderer to imagery raster layer.
+                val renderer = BlendRenderer(
+                    elevationRaster = elevationRaster,
+                    outputMinValues = listOf(9.0),
+                    outputMaxValues = listOf(255.0),
+                    sourceMinValues = emptyList(),
+                    sourceMaxValues = emptyList(),
+                    noDataValues = emptyList(),
+                    gammas = emptyList(),
+                    colorRamp = null,
+                    altitude = altitude,
+                    azimuth = azimuth,
+                    slopeType = slopeType ?: SlopeType.None
+                )
+
+                imageryRasterLayer.renderer = renderer
+                arcGISMap.setBasemap(imageryBasemap)
+            }
```

Comment:
The `renderer` could be moved out of the ` if (colorRamp != null)` check: 

```suggestion
        val colorRamp = createColorRampFromPresetIndex(selectedColorRampPresetIndex)
        val renderer = BlendRenderer(
            elevationRaster = elevationRaster,
            outputMinValues = listOf(9.0),
            outputMaxValues = listOf(255.0),
            sourceMinValues = emptyList(),
            sourceMaxValues = emptyList(),
            noDataValues = emptyList(),
            gammas = emptyList(),
            colorRamp = colorRamp,
            altitude = altitude,
            azimuth = azimuth,
            slopeType = slopeType
        )
        if (colorRamp != null) {
            // When a color ramp is applied, set the basemap to the elevation raster
            elevationRasterLayer.renderer = renderer
            arcGISMap.setBasemap(elevationBasemap)
        } else {
            // Since no ColorRamp is selected, apply BlendRenderer to imagery raster layer.
            imageryRasterLayer.renderer = renderer
            arcGISMap.setBasemap(imageryBasemap)
        }
```